dipy/io/peaks.py

```diff
@@ -0,0 +1,237 @@
+
+    Returns
+    -------
+    pam : PeaksAndMetrics object
+    """
+
+    if os.path.splitext(fname)[1] != '.pam5':
+        raise IOError('This function supports only PAM5 (HDF5) files')
+
+    if TABLES_LESS_3_0:
+        func_open_file = tables.openFile
+    else:
+        func_open_file = tables.open_file
+
+    f = func_open_file(fname, 'r')
+
+    pam = PeaksAndMetrics()
+
+    pamh = f.root.pam
+
+    version = f.root.version[0].decode()
+
+    try:
+        affine = pamh.affine[:]
+    except tables.NoSuchNodeError:
+        affine = None
+
+    peak_dirs = pamh.peak_dirs[:]
+    peak_values = pamh.peak_values[:]
+    peak_indices = pamh.peak_indices[:]
+
+    try:
+        shm_coeff = pamh.shm_coeff[:]
+    except tables.NoSuchNodeError:
+        shm_coeff = None
+
+    sphere_vertices = pamh.sphere_vertices[:]
+
+    try:
+        odf = pamh.odf[:]
+    except tables.NoSuchNodeError:
+        odf = None
+
+    pam.affine = affine
+    pam.peak_dirs = peak_dirs
+    pam.peak_values = peak_values
+    pam.peak_indices = peak_indices
+    pam.shm_coeff = shm_coeff
+    pam.sphere = Sphere(xyz=sphere_vertices)
+    pam.B = pamh.B[:]
+    pam.total_weight = pamh.total_weight[:][0]
+    pam.ang_thr = pamh.ang_thr[:][0]
+    pam.gfa = pamh.gfa[:]
+    pam.qa = pamh.qa[:]
+    pam.odf = odf
+
+    f.close()
+
+    if verbose:
+        print('PAM5 version')
+        print(version)
+        print('Affine')
+        print(pam.affine)
+        print('Dirs shape')
+        print(pam.peak_dirs.shape)
+        print('SH shape')
+        if pam.shm_coeff is not None:
+            print(pam.shm_coeff.shape)
+        else:
+            print('None')
+        print('ODF shape')
+        if pam.odf is not None:
+            print(pam.odf.shape)
+        else:
+            print('None')
+        print('Total weight')
+        print(pam.total_weight)
+        print('Angular threshold')
+        print(pam.ang_thr)
+        print('Sphere vertices shape')
+        print(pam.sphere.vertices.shape)
+
+    return pam
+
+
+def save_peaks(fname, pam, affine=None, verbose=False):
+    """ Save all important attributes of object PeaksAndMetrics in a PAM5 file
+    (HDF5).
+
+    Parameters
+    ----------
+    fname : string
+        Filenam of PAM5 file
+    pam : PeaksAndMetrics
+        Object holding peak_dirs, shm_coeffs and other attributes
+    affine : array
+        The 4x4 matrix transforming the date from native to world coordinates.
+        PeaksAndMetrics should have that attribute but if not it can be
+        provided here. Default None.
+    verbose : bool
+        Print summary information about the saved file.
+    """
+
+    if os.path.splitext(fname)[1] != '.pam5':
+        raise IOError('This function saves only PAM5 (HDF5) files')
+
+    if not (hasattr(pam, 'peak_dirs') and hasattr(pam, 'peak_values') and
+            hasattr(pam, 'peak_indices')):
+
+        raise ValueError('Cannot save object without peak_dirs/values/indices')
```

Comment:
This error string might be confusing. How about: "Cannot save object without peak_dirs, peak_values and peak_indices" (to make it clear that all three are required).